python/ray/dashboard/modules/job/utils.py

```diff
@@ -302,3 +301,9 @@ async def find_jobs_by_job_ids(
             for job_info, submission_id in zip(job_infos, job_submission_ids)
         },
     }
+
+
+def close_logger_file_descriptor(logger_instance):
+    for handler in logger_instance.handlers:
+        handler.close()
+        logger_instance.removeHandler(handler)
```

Comment:
I'm fine with this approach, but note that this will cause all future use of this logger becoming no-ops. Based on this [post](https://stackoverflow.com/a/15474586) It seems like just call `handler.close()` is enough to close out the file descriptor. Maybe we can refactor a `close_fd_after_logs()` and that calls `.info` then close the handler so the logger can be used for future?

Reply:
Hey @GeneDer. I have made following changes to the PR. 

- I have moved to closing of logger to finally block so that info/error can be logged in between if required.
- Moved logger to global context of submit_job function instead of try block so that same logger can be used in exception as well.
- Warning in exception block was being routed to wrong logger, moved it to job logger for easy debugging.
- Added a unit test for the same. Also handler.close is not enough we have to remove handler as well. Tested this as well.


Please let me know your views :)

Reply:
One problem that I can see  with  `close_fd_after_logs() and that calls .info` is that we have to offer warn, error etc as well. Then we would be making a custom wrapper over logger to provide all functionalities.